### PR TITLE
Add ability to change the number of concurrent downloads

### PR DESCRIPTION
**Pull request by needlesslygrim**

### CODE REVIEW

**Comment by needlesslygrim** [comment]
Opening this as a draft PR for two reasons:
1. It's not finished yet, only the actual settings page is updated
2. The switch and text input are using the default Material 2 design

**Comment by needlesslygrim** [comment]
UI screenshots:

image

image

File: pubspec.lock [hunk 1]
```diff
@@ -721,10 +721,17 @@ packages:
...
       sha256: "3923c89304b715fb1eb6423f017651664a03bf5f4b29983627c4da791f74a4ec"
       url: "https://pub.dev"
     source: hosted
     version: "14.2.1"
+=======
+      sha256: b3d56ff4341b8f182b96aceb2fa20e3dcb336b9f867bc0eafc0de10f1048e957
+      url: "https://pub.dev"
+    source: hosted
+    version: "13.0.0"
+>>>>>>> Stashed changes
```
Comment (micfong-z) [inline pubspec.lock]: Unresolved git diff here.

**Review by micfong-z — CHANGES_REQUESTED** [review]

File: pubspec.lock (hunk 1 above)
Comment (needlesslygrim) [reply pubspec.lock]: Oops, I'll fix that!

Comment (needlesslygrim) [reply pubspec.lock]: Should be fixed now.

**Comment by micfong-z** [comment]
Trust me, I'll definitely review this soon™️

**Review by micfong-z — APPROVED** [review]
LGTM functionality wise.

I'll probably do some UI dirty work in a few month's time.

**Pull request merged**